### PR TITLE
Fix issues caused by float imprecision

### PR DESCRIPTION
**Pull request by jordan-brough**
Because, e.g., `8.21.to_d != 8.21`

This was causing `validate_authorization` to fail for certain numbers because it didn't think enough funds were available.

### CODE REVIEW

File: app/models/spree/payment_method/store_credit.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def authorize(amount_in_cents, store_credit, gateway_options = {})
       else
         action = -> (store_credit) {
           store_credit.authorize(
-            amount_in_cents / 100.0,
+            amount_in_cents / 100.0.to_d,
```
Comment (magnusvk) [inline app/models/spree/payment_method/store_credit.rb:22]: Wouldn't it be sufficient to just have `100.to_d`? In fact, it should be (theoretically) safer as the integer `100` is exact precision whereas the float `100.0` isn't.

**Comment by magnusvk** [comment]
@jordan-brough :+1: Looks like the test failure is unrelated. I assume tests were green locally?

Can we get this shipped to production before it gets merged in the fork perhaps?

**Comment by jordan-brough** [comment]
@magnusvk yup the specs pass locally. merged here and added to our 2-2-dev branch also. made a ticket to fix the travis specs.

**Comment by magnusvk** [comment]
Awesome.

**Pull request merged**